### PR TITLE
mirage-kv 2.0.0

### PR DESCRIPTION
**Pull request by hannesm**
this upgrades tar-mirage to the mirage-kv 2.0.0 API :D

### CODE REVIEW

**Comment by hannesm** [comment]
while doing this port, I noticed that this is unlikely to work on Solo5, where `block_read` expects at most one sector to be read -- while in tar-mirage, `BLOCK.read` gets a single huge buffer passed. do other block backends (unix/xen) have the same semantics as solo5-block? .oO(feels like a revision of mirage-block needs to be done /cc @g2p @samoht @mato)

File: mirage/tar_mirage.ml [hunk 1]
```diff
@@ -129,18 +122,97 @@ module Make_KV_RO (BLOCK : Mirage_block_lwt.S) = struct
...
+    in
+    Lwt.return r
+
+  let digest t key =
+    get t key >|= function
+    | Error e -> Error e
+    | Ok data -> Ok (Digest.string data)
+
+  (* Compare filenames without a leading / or ./ *)
+  let trim_slash x =
```
Comment (avsm) [inline mirage/tar_mirage.ml]: Is this function something that can be replaced with `Fpath.normalize` use?

Comment (hannesm) [reply mirage/tar_mirage.ml]: We're using [Mirage_kv.Key](https://mirage.github.io/mirage-kv/mirage-kv/Mirage_kv/Key/index.html), where I can't find such a function. But maybe worth an addition!

**Comment by mato** [comment]
> feels like a revision of mirage-block needs to be done /cc @g2p @samoht @mato

See https://github.com/Solo5/solo5/issues/325#issuecomment-465483894 for what needs to be done in Solo5 to support block operations in multiples of the block size.

**Comment by hannesm** [comment]
ok, thanks. the block read/write did not change in this PR, and should be addressed in other PRs. IMHO this is ready to be merged + released to unbreak tar-mirage with mirage 3.5.0 (which uses mirage-kv 2.0.0). any concerns about merging this PR?

**Comment by samoht** [comment]
LGTM

**Comment by hannesm** [comment]
how can we move forward here? `tar-mirage` is atm not compatible with mirage 3.5.0 due to the changes in mirage-kv. IMHO it'd be great to have this merged and released //cc @djs55

**Comment by djs55** [comment]
@samoht thanks for taking care of this!

**Pull request merged**